### PR TITLE
support unexport field

### PR DESCRIPTION
**Pull request by go-wyvern**
fix bug: https://github.com/goplus/gop/issues/558

### CODE REVIEW

File: exec/bytecode/struct.go [hunk 1]
```diff
@@ -177,8 +178,8 @@ func makeStruct(typStruct reflect.Type, arity int, p *Context) {
 	}
 	v := reflect.New(typStruct).Elem()
 	for i := 0; i < n; i += 2 {
-		fieldName := args[i].(string)
-		v.FieldByName(fieldName).Set(reflect.ValueOf(args[i+1]))
+		index := args[i].([]int)
```
Comment (xushiwei) [inline exec/bytecode/struct.go:181]: I think args[i] is `int` better than `[]int`

File: exec/bytecode/struct.go [hunk 2]
```diff
@@ -177,8 +178,8 @@ func makeStruct(typStruct reflect.Type, arity int, p *Context) {
...
+		v.FieldByIndex(index).Set(reflect.ValueOf(args[i+1]))
```
Comment (xushiwei) [inline exec/bytecode/struct.go:182]: And then `FieldByIndex(index)` will be `Field(index)`

File: exec/bytecode/struct_test.go [hunk 3]
```diff
@@ -41,9 +41,11 @@ func TestStruct(t *testing.T) {
...
 			b.DefineVar(v)
-			b.Push("Name")
+			b.Push(0)
+			b.MakeArray(reflect.TypeOf([]int{0}), 1)
 			b.Push("bar")
-			b.Push("Age")
+			b.Push(1)
+			b.MakeArray(reflect.TypeOf([]int{1}), 1)
 			b.Push(30)
 			b.Struct(reflect.TypeOf(p), 2)
```
Comment (xushiwei) [inline exec/bytecode/struct_test.go:50]:
```go
foo := Foo{
   A: 1,
   B: "Hi",
   C: Bar{D: 1},
}
```

And we can't use:

```go
foo := Foo{
    A: 1,
    B: "Hi",
    C.D: 1,
}
```

So, we don't need a `[]int` to indicate index, `int` is enough.

Comment (go-wyvern) [reply exec/bytecode/struct_test.go:50]: Oh! Very good example, thanks~!

**Pull request merged**